R/as_gt.R

```diff
@@ -0,0 +1,144 @@
+#  Copyright (c) 2024 Merck & Co., Inc., Rahway, NJ, USA and its affiliates.
+#  All rights reserved.
+#
+#  This file is part of the simtrial program.
+#
+#  simtrial is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#' Convert summary table to a gt object
+#'
+#' @param x A summary object of a fixed or group sequential design.
+#' @param ... Additional arguments (not used).
+#'
+#' @return A gt table.
+#'
+#' @export
+as_gt <- function(x, ...) {
+  UseMethod("as_gt", x)
+}
```

Comment:
I confirmed that we also export `as_gt()` in gsDesign2

https://github.com/Merck/gsDesign2/blob/a77b22f114eba302f199e11941e649a4f6aee739/NAMESPACE#L13

Reply:
Exactly. If we have `as_gt` in `simtrial`, we should have `... |> simtrial::as_gt()` to distinguish between `gsDesign2::as_gt()`.

Do you happen to have an alternative way? such as re-naming `as_gt`?

Reply:
No, I can't think of a better way. While you could rename `as_gt()`, I think it is more consistent to keep the same name.

Reply:
Agree!